spec/controller/authors_controller_spec.rb

```diff
@@ -0,0 +1,107 @@
+require "rails_helper"
+
+RSpec.describe AuthorsController, type: :controller do
+  let(:user) { User.create!(name: "User", email: "user@example.com", password: "password", gender: :male, date_of_birth: Date.new(1990,1,1)) }
+  let(:author) { Author.create!(name: "Nam Cao") }
+  let(:publisher) { Publisher.create!(name: "NXB Kim Đồng") }
+  let!(:book) { Book.create!(title: "Chí Phèo", author: author, publisher: publisher, total_quantity: 5) }
+
+  before do
+    session[:user_id] = user.id
+  end
+
+  describe "GET #show" do
+    context "when author exists" do
+      it "assigns the author and renders show" do
+        get :show, params: { id: author.id }
+        expect(assigns(:author)).to eq(author)
+        expect(response).to be_successful
```

Comment:
Tương tự cho controller này, tách mỗi it chỉ có 1 expect thôi e nha
Tương tự cho các action khác